test/conformance/enqueue/urEnqueueUSMFill2D.cpp

```diff
@@ -1,2 +1,288 @@
+    int *ptr{nullptr};
+};
+
+static std::vector<testParametersFill2D> test_cases{
+    /* Everything set to 1 */
+    {1, 1, 1, 1},
+    /* Height == 1 && Pitch > width && pattern_size == width*/
+    {1024, 256, 1, 256},
+    /* Height == 1 && Pitch > width && pattern_size < width*/
+    {1024, 256, 1, 4},
+    /* Height == 1 && Pitch > width && width != power_of_2 && pattern_size < width */
+    {1024, 57, 1, 4},
+    /* Height == 1 && Pitch == width && pattern_size < width */
+    {1024, 1024, 1, 256},
+    /* Height == 1 && Pitch == width && pattern_size == width */
+    {1024, 1024, 1, 1024},
+    /* Height > 1 && Pitch > width && pattern_size == 1 */
+    {1024, 256, 256, 1},
+    /* Height > 1 && Pitch > width && pattern_size == width */
+    {1024, 256, 256, 256},
+    /* Height > 1 && Pitch > width && pattern_size == width * height */
+    {1024, 256, 256, 256 * 256},
+    /* Height > 1 && Pitch == width + 1 && pattern_size == 2 */
+    {234, 233, 23, 2},
+    /* Height == 1 && Pitch == width + 1 && pattern_size == 128 */
+    {234, 233, 1, 128},
+    /* Height != power_of_2 && Pitch == width + 1 && pattern_size == 128 */
+    {234, 233, 35, 128}};
+
+UUR_TEST_SUITE_P(urEnqueueUSMFill2DTestWithParam,
+                 testing::ValuesIn(test_cases),
+                 printFill2DTestString<urEnqueueUSMFill2DTestWithParam>);
+
+TEST_P(urEnqueueUSMFill2DTestWithParam, Success) {
+
+    ur_event_handle_t event = nullptr;
+
+    ASSERT_SUCCESS(
+        urEnqueueUSMFill2D(queue, ptr, pitch, pattern_size, pattern.data(),
+                           width, height, 0, nullptr, &event));
+    EXPECT_SUCCESS(urQueueFlush(queue));
+
+    ASSERT_SUCCESS(urEventWait(1, &event));
+    ur_event_status_t event_status;
+    EXPECT_SUCCESS(urEventGetInfo(event, UR_EVENT_INFO_COMMAND_EXECUTION_STATUS,
+                                  sizeof(ur_event_status_t), &event_status,
+                                  nullptr));
+    ASSERT_EQ(UR_EVENT_STATUS_COMPLETE, event_status);
+    EXPECT_SUCCESS(urEventRelease(event));
+
+    ASSERT_NO_FATAL_FAILURE(verifyData());
+}
+
+struct urEnqueueUSMFill2DNegativeTest : uur::urQueueTest {
+    void SetUp() override {
+        UUR_RETURN_ON_FATAL_FAILURE(uur::urQueueTest::SetUp());
+
+        bool device_usm{false};
+        ASSERT_SUCCESS(
+            urDeviceGetInfo(device, UR_DEVICE_INFO_USM_DEVICE_SUPPORT,
+                            sizeof(bool), &device_usm, nullptr));
+
+        if (!device_usm) {
+            GTEST_SKIP() << "Device USM is not supported";
+        }
+
+        ur_usm_mem_flags_t flags{};
+        ASSERT_SUCCESS(
+            urUSMDeviceAlloc(context, device, &flags, allocation_size, 0,
+                             reinterpret_cast<void **>(&ptr)));
+    }
+
+    void TearDown() override {
+        if (ptr) {
+            EXPECT_SUCCESS(urUSMFree(context, ptr));
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(uur::urQueueTest::TearDown());
+    }
+
+    const size_t pitch = 16;
```

Comment:
I would avoid making member variables `const` : [see here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-constref)

Reply:
make it `static constexpr` instead